lib/tbd/psi.rb

```diff
@@ -1558,8 +1569,23 @@ def process(model = nil, argh = {})
     faces(walls   , edges)
     faces(shades  , edges)
 
+    # Purge existing KIVA objects from model.
+    if argh[:reset_kiva]
+      kva = false
+      kva = true unless model.getSurfacePropertyExposedFoundationPerimeters.empty?
+      kva = true unless model.getFoundationKivas.empty?
+
+      if kva
+        if argh[:gen_kiva]
+          resetKIVA(model, "Foundation")
+        else
+          resetKIVA(model, "Ground")
+        end
+      end
+    end
```

Comment:
If _measure.rb_'s **run** method is the (higher level) calling method, then **resetKIVA** checks here are redundant (TBD can be called directly as a gem method).